### PR TITLE
Live: comments for pipeline channel rule

### PR DESCRIPTION
**Pull request by FZambia**
**What this PR does / why we need it**:

Add detailed comments which summarize current Pipeline behavior.

### CODE REVIEW

File: pkg/services/live/pipeline/pipeline.go [hunk 1]
```diff
@@ -120,17 +120,45 @@ type SubscribeAuthChecker interface {
...
-	SubscribeAuth   SubscribeAuthChecker
-	Subscribers     []Subscriber
-	DataOutputters  []DataOutputter
-	Converter       Converter
+	// OrgId this rule belongs to.
+	OrgId int64
+	// Pattern is a pattern for a channel which when matched result into rule execution
+	// during Subscribe or Publish operations. This is very similar to HTTP router functionality.
+	// We use a modified version of github.com/julienschmidt/httprouter for pattern matching logic
+	// (see tree package).
```
Comment (ArturWierzbicki) [inline pkg/services/live/pipeline/pipeline.go:130]: should we add an example here? especially one which would be handled differently than in `julienschmidt/httprouter` (if that's even a possibility)

Comment (FZambia) [reply pkg/services/live/pipeline/pipeline.go:130]: Yeah, good note – this requires some effort though. I personally don't know the exact differences yet (only some of them), there is a readme https://github.com/grafana/grafana/blob/main/pkg/services/live/pipeline/tree/readme.md with some motivation which has some additional information regarding this.

**Review by ryantxu — APPROVED** [review]
:heart:

File: pkg/services/live/pipeline/pipeline.go (hunk 1 above)
Comment (FZambia) [reply pkg/services/live/pipeline/pipeline.go]: Addressed this a bit in ff6a6c1a47609f0f39f84c028c6bf4d523b27b10 - I think can be revisited and extended more on the later stages.

**Pull request merged**